serial.go

```diff
@@ -84,12 +85,12 @@ type Config struct {
 	// XONFlowControl bool
 
 	// CRLFTranslate bool
-	// TimeoutStuff int
+	ReadTimeout uint32
```

Comment:
I really think this should be a time.Duration.  time.Duration is the standard for intervals in Go and it has well understood units (nanoseconds) and conversion to milliseconds or 100 milliseconds is one easy operation.  For example, if this were to stay uint32 milliseconds, I think it should get an explicit "_ms" suffix and a comment saying the units.  time.Duration gets rid of that need.

Also please move this higher up in the struct and add a line of documentation about which kind of timeout it is.  My understanding of your intention is that this is an overall timeout NOT an interval timeout.  A comment would be very helpful to a user of the library.